### PR TITLE
Changed consensus type E2E_Test from solo to raft

### PR DESCRIPTION
**Pull request by RohanShrothrium**
Signed-off-by: rohanshrothrium <shrothriumrohan@gmail.com>
This PR introduces changes to the e2e_test and now uses raft consensu for ordering instead of solo ordering.

### CODE REVIEW

**Comment by dexhunter** [comment]
Hmm...why a lot of secret keys are changing?

**Comment by yeasy** [comment]
> Hmm...why a lot of secret keys are changing?

+1. 
No need to regenerate those credentials?

**Comment by RohanShrothrium** [comment]
> Hmm...why a lot of secret keys are changing?

@yeasy and @dexhunter, I'll use the previously used certificates and update the PR.

**Comment by RohanShrothrium** [comment]
I have reverted back to the previous certs! Please review the same!

**Comment by dexhunter** [comment]
@RohanShrothrium  What are those certificates under the `test/fixtures/e2e_raft/crypto-config/.svn/`?

**Comment by RohanShrothrium** [comment]
> @RohanShrothrium What are those certificates under the `test/fixtures/e2e_raft/crypto-config/.svn/`?

Ohh I downloaded the previous version of the `crypto-config` and `channel-artifacts` using svn so that was leftover. I'll remove it.

File: test/fixtures/network-raft.json [hunk 1]
```diff
@@ -28,6 +28,21 @@
         }
       }
     },
+    "orderer5.example.com": {
```
Comment (dexhunter) [inline test/fixtures/network-raft.json:31]: the naming here is a bit confusing, why starting from `orderer5`?

Comment (RohanShrothrium) [reply test/fixtures/network-raft.json:31]: There are 5 orderers and I have added only the credentials of `orderer` and `orderer5` in the `network.json` file.

**Comment by dexhunter** [comment]
@RohanShrothrium The error message is 
```grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "failed to connect to all addresses"
	debug_error_string = "{"created":"@1602679823.669218145","description":"Failed to pick subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":4133,"referenced_errors":[{"created":"@1602679823.669214245","description":"failed to connect to all addresses","file":"src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc","file_line":397,"grpc_status":14}]}"
```
which is different from the timeout error. I found [this answer](https://stackoverflow.com/questions/59823424/grpc-14-unavailable-failed-to-connect-to-all-addresses) indicating that gRPC fails to connect to all the address because no server is running on the address. I think you need to add some timeout before the invocation.

**Comment by RohanShrothrium** [comment]
@dexhunter The error is because many integration tests use the `utils.py`, so I'll have to make changes to the rest of the integration tests too. I'll update the PR by the earliest.

**Comment by RohanShrothrium** [comment]
@dexhunter the CI fails in the e2e_test but when I run it locally it's passing. I don't know why exactly this is happening.

**Review by yeasy — DISMISSED** [review]

**Comment by yeasy** [comment]
@RohanShrothrium If this is applicable for both HLF v1.4 and v2.x, please add a new patchset to https://github.com/hyperledger/fabric-sdk-py/tree/release-1.0 too.

**Comment by dexhunter** [comment]
The [current error message](https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=24243&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=2156) is 

```
 info: "error applying config update to existing channel \'businesschannel\': error authorizing update: error validating ReadSet: proposed update requires that key [Group]  /Channel/Application be at version 0, but it is currently at version 1"
```

**Comment by RohanShrothrium** [comment]
@dexhunter and @yeasy can you help me fix the issue with the CI failing. I'm not sure why this is happening.

**Comment by dexhunter** [comment]
@RohanShrothrium Can you rebase to the latest commit of the master branch?

**Comment by dexhunter** [comment]
@RohanShrothrium Can you check [the error log](https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=25031&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=2254)? It seems to be a configuration problem.

File: test/integration/e2e_test.py [hunk 2]
```diff
@@ -698,9 +698,8 @@ def test_in_sequence(self):
 
         self.client.new_channel(SYSTEM_CHANNEL_NAME)
 
-        loop.run_until_complete(self.get_channel_config_with_orderer())
```
Comment (dexhunter) [inline test/integration/e2e_test.py:701]: just wonder why remove this test here?

File: test/integration/e2e_test.py [hunk 3]
```diff
@@ -698,9 +698,8 @@ def test_in_sequence(self):
...
-
         loop.run_until_complete(self.channel_create())
+        time.sleep(5)
```
Comment (dexhunter) [inline test/integration/e2e_test.py:702]: also why there is a sleep added here?

File: test/integration/utils_raft.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+# This should be deprecated, and use client.get_user() API instead
+def get_peer_org_user(org, user, state_store):
+    """Loads the requested user for a given peer org
+        and returns a user object.
+    """
+
+    peer_user_base_path = os.path.join(
+        os.getcwd(),
+        'test/fixtures/e2e_cli/crypto-config/peerOrganizations/{0}'
```
Comment (dexhunter) [inline test/integration/utils_raft.py:132]: shouldn't it be corresponding to the `e2e_raft` ones?

File: test/integration/utils_raft.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+def get_orderer_org_user(org='example.com', user='Admin', state_store=None):
+    """Loads the admin user for a given orderer org and
+        returns an user object.
+        Currently, orderer org only has Admin
+
+    """
+    msp_path = os.path.join(
+        os.getcwd(),
+        'test/fixtures/e2e_cli/crypto-config/ordererOrganizations/'
```
Comment (dexhunter) [inline test/integration/utils_raft.py:159]: same here

File: test/integration/utils_raft.py (hunk 4 above)
Comment (RohanShrothrium) [reply test/integration/utils_raft.py:132]: This function is not being used by the test, so I haven't changed it.

File: test/integration/utils_raft.py (hunk 5 above)
Comment (RohanShrothrium) [reply test/integration/utils_raft.py:159]: This function is also not being used by the test, so I haven't changed it.

File: test/integration/e2e_test.py (hunk 3 above)
Comment (RohanShrothrium) [reply test/integration/e2e_test.py:702]: In the raft consensus, the channel creation takes some time and I was getting errors so I added the sleep function.

**Pull request closed without merging**